wpilibc/src/main/native/cpp/Joystick.cpp

```diff
@@ -183,7 +179,17 @@ bool Joystick::GetTrigger(JoystickHand hand) const {
  * @return The state of the top button.
  */
 bool Joystick::GetTop(JoystickHand hand) const {
-  return GetRawButton(m_buttons[kTopButton]);
+  return GetRawButton(static_cast<int>(Button::kTop));
+}
+
+Joystick* Joystick::GetStickForPort(int port) {
+  static std::array<Joystick*, DriverStation::kJoystickPorts> joysticks{};
+  Joystick* stick = joysticks[port];
+  if (stick == nullptr) {
+    stick = new Joystick(port);
+    joysticks[port] = stick;
+  }
+  return stick;
```

Comment:
This is a memory leak.  We should at least use unique_ptr in the array if we're doing this.

Reply:
I'd honestly rather not implement this function at all. It's not worth the effort because teams really, really shouldn't be using this and it would get removed next year following our typical deprecation cycle anyway.